bson/src/test/unit/org/bson/codecs/pojo/entities/conventions/CreatorConstructorModel.java

```diff
@@ -19,24 +19,28 @@
 import org.bson.codecs.pojo.annotations.BsonCreator;
 import org.bson.codecs.pojo.annotations.BsonProperty;
 
+import java.util.List;
+
 public final class CreatorConstructorModel {
-    private final Integer integerField;
+    private final List<Integer> integersField;
     private String stringField;
     public long longField;
 
     @BsonCreator
-    public CreatorConstructorModel(@BsonProperty("integerField") final Integer integerField) {
-        this.integerField = integerField;
+    public CreatorConstructorModel(@BsonProperty("integersField") final List<Integer> integerField,
+                                   @BsonProperty("longField") final long longField) {
```

Comment:
This doesn't seem to test the assignability support.  To test that, can you add a test where parameter is long and the property is Long, or parameter is int and property is long?  Are there any other interesting cases?

Reply:
List is an interesting case as its boxed to ArrayList by Type Data - failing the comparison.   `long` is also boxed - so that would fail without the assignability check.

The issue is we are comparing against the PropertyModel's TypeData against a raw type.  However, there is no need to try and create a new instance of the TypeData and potentially discover all the parameter types.

Reply:
OK, I got it.